design/src/rfcs/rfc0007_split_release_process.md

```diff
@@ -0,0 +1,143 @@
+RFC: Split Release Process
+==========================
+
+> Status: RFC
+
+At the time of writing, the `aws-sdk-rust` repository is used exclusively
+for the entire release process of both the Rust runtime crates from `smithy-rs` as
+well as the AWS runtime crates and the AWS SDK. This worked well when `smithy-rs` was
+only used for the AWS SDK, but now that it's also used for server codegen, there
+are issues around publishing the server-specific runtime crates since they don't
+belong to the SDK.
+
+This RFC proposes a new split-release process so that the entire `smithy-rs` runtime
+can be published separately before the AWS SDK is published.
+
+Terminology
+-----------
+
+- **Smithy Runtime Crate**: A crate that gets published to crates.io and supports
+  the code generated by `smithy-rs`. These crates don't provide any SDK-only functionality.
+  These crates can support client and/or server code, and clients or servers may use
+  only a subset of them.
+- **AWS Runtime Crate**: A crate of SDK-specific code that supports the code generated
+  by the `aws/codegen` module in `smithy-rs`. These also get published to crates.io.
+- **Publish-ready Bundle**: A build artifact that is ready to publish to crates.io without
+  additional steps (such as running the publisher tool's `fix-manifests` subcommand). Publishing
+  one group of crates before another is not considered an additional step for this definition.
+- **Releaser**: A developer, automated process, or combination of the two that performs the actual release.
+
+Requirements
+------------
+
+At a high level, the requirements are: publish from both `smithy-rs` and `aws-sdk-rust`
+while preserving our current level of confidence in the quality of the release. This
+can be enumerated as:
+
+1. All Smithy runtime crates must be published together from `smithy-rs`
+2. AWS runtime crates and the SDK must be published together from `aws-sdk-rust`
+3. CI on `smithy-rs` must give confidence that the Smithy runtime crates,
+   AWS runtime crates, and SDK are all at the right quality bar for publish.
+4. CI on the `aws-sdk-rust` repository must give confidence that the AWS SDK and its
+   runtime crates are at the right quality bar for publish. To do this successfully,
+   it must run against the exact versions of the Smithy runtime crates the code was
+   generated against _both before AND after they have been published to crates.io_.
+
+Background: How Publishing Worked Before
+----------------------------------------
+
+The publish process to crates.io relied on copying all the Smithy runtime crates
+into the final `aws-sdk-rust` repository. Overall, the process looked as follows:
+
+1. `smithy-rs` generates a complete `aws-sdk-rust` source bundle at CI time
+2. The releaser copies the generated bundle over to `aws-sdk-rust`
```

Comment:
Maybe replace releaser with engineer to clarify that it's a human and not an automated job

Reply:
"Releaser" was defined above as either an engineer or an automated process (or a combination of the two), so I think this is fine.